libraries/libfc/include/fc/crypto/packhash.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <fc/io/raw.hpp>
+
+namespace fc {
+
+template<typename Encoder, typename... T>
+requires ( sizeof...(T) > 0 )
+auto packhash(const T&... t) {
+   Encoder e;
+   raw::pack(e,t...);
+   return e.result();
+}
+
+template<typename Derived>
+struct add_packhash_to_hash {
+   template<typename... T>
+   requires ( sizeof...(T) > 0 )
+   static auto packhash(const T&... args) {
+      return fc::packhash<typename Derived::encoder>(args...);
+   }
+};
+
+}
```

Comment:
I am willing to hear that we should just copy/paste `packhash()` in to every hash class instead of doing the funky CRTP which injects it. I felt like `packhash()` being right alongside `hash()` was important. (i.e. `sha256::hash()` and `sha256::packhash()`)

Reply:
I like it as it is personally.